### ISSUE TITLE
Initialization errors are reduced to warning and stack is not shown

### ISSUE THREAD

**Issue by Domiii** [storybookjs/storybook#16531]
**Describe the bug**
If initialization errors out, storybook does very little to provide the most basic information as to what went wrong. The error is shown as a _**warning**_ (not sure when this would ever make sense?), and the `stack` is not available at all.

**To Reproduce**
Create any story and `throw new Error()` anywhere during initialization (e.g. in the story file before the story begins).

**System**

```
Environment Info:

  System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
  Binaries:
    Node: 16.5.0 - ~\AppData\Local\Volta\tools\image\node\16.5.0\node.EXE
    Yarn: 1.22.10 - ~\AppData\Local\Volta\tools\image\yarn\1.22.10\bin\yarn.CMD
    npm: 7.19.1 - ~\AppData\Local\Volta\tools\image\node\16.5.0\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (95.0.1020.30)
  npmPackages:
    @storybook/addon-actions: ^6.4.0-alpha.3 => 6.4.0-alpha.22
    @storybook/addon-essentials: ^6.4.0-alpha.3 => 6.4.0-alpha.22
    @storybook/addon-links: ^6.4.0-alpha.3 => 6.4.0-alpha.22
    @storybook/addon-postcss: ^2.0.0 => 2.0.0
    @storybook/builder-webpack5: ^6.4.0-alpha.3 => 6.4.0-alpha.22
    @storybook/manager-webpack5: ^6.4.0-alpha.3 => 6.4.0-alpha.22
    @storybook/react: ^6.4.0-alpha.3 => 6.4.0-alpha.22
```

**Additional context**

1. Please do not log an **_error_** as a `warning`.
2. Please **_never_** hide the original `stack` of an error, lest you plan to rob developers off one of their most fundamental quality-of-life features.
   * If you want to nest errors and/or attach additional information to the thrown error, please consider [`esnext` nested error ("error cause")](https://github.com/tc39/proposal-error-cause).
   * (...or any of the hacks that emulate this very basic error propagation tool)

**Comment by shilman** [comment]
I have repro'ed this in both 6.3.x and 6.4.0-beta.x, though in the latter it shows a stack trace. I agree this is a pretty bad DX, and we'll try to fix it before 6.4 is released. Thanks for reporting!

**Comment by shilman** [comment]
Ta-da!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.4.0-beta.27 containing PR #16537 that references this issue. Upgrade today to the `@next` NPM tag to try it out!

```
npx sb upgrade --prerelease
```

Closing this issue. Please re-open if you think there's still more to do.
